hypothesis-python/src/hypothesis/extra/github_actions.py

```diff
@@ -0,0 +1,156 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Copyright the Hypothesis Authors.
+# Individual contributors are listed in AUTHORS.rst and the git log.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+import shutil
+import tempfile
+from os import getenv
+
+import requests
+
+from hypothesis.configuration import mkdir_p, storage_directory
+from hypothesis.database import DirectoryBasedExampleDatabase
+
+
+class GitHubArtifactDatabase(DirectoryBasedExampleDatabase):
+    """
+    A directory-based database loaded from a GitHub Actions artifact.
+
+    This is useful for sharing example databases between CI runs and developers, allowing the latter
+    to get read-only access to the former. In most cases, this will be used through the
+    :class:`~hypothesis.database.MultiplexedDatabase`, by combining a local directory-based database
+    with this one. For example:
+
+    .. code-block:: python
+
+        local = DirectoryBasedExampleDatabase(".hypothesis/examples")
+        shared = GitHubArtifactDatabase("user", "repo")
+
+        settings.register_profile("ci", database=local)
+        settings.register_profile("dev", database=MultiplexedDatabase(local, shared))
+        settings.load_profile("ci" if os.environ.get("CI") else "dev")
```

Comment:
The dream here would be to have it work locally as well, so that if you run your tests on a laptop or in Codespaces or whatever, it downloads the artifact there too... not sure how much harder that would make it though, and I'm very happy for you to rule that out of scope for this PR.

Reply:
I'm confused, how does this differ from the setup we have already?

Reply:
🤦‍♂️, I think it *does* do this already, sorry.  I was thinking about the Actions config in https://github.com/Zac-HD/hypofuzz/issues/14#issuecomment-1423910866 but that's just treating the local database as an artifact 👍